hbase-server/src/main/java/org/apache/hadoop/hbase/master/MobFileCleanerChore.java

```diff
@@ -0,0 +1,289 @@
+                  LOG.warn("Found old store file with no MOB_FILE_REFS: {} - "
+                    + "can not proceed until all old files will be MOB-compacted.", pp);
+                  return;
+                } else if (mobRefData == null && bulkloadMarkerData != null) {
+                  LOG.info("Skipping file without MOB references (bulkloaded file):{}", pp);
+                  continue;
+                }
+                if (mobRefData.length > 1) {
+                  // if length = 1 means NULL, that there are no MOB references
+                  // in this store file, but the file was created by new MOB code
+                  String[] mobs = new String(mobRefData).split(",");
+                  LOG.debug("Found: {} mob refs: ", mobs.length, Arrays.toString(mobs));
+                  regionMobs.addAll(Arrays.asList(mobs));
+                }
+              }
+            } catch (FileNotFoundException e) {
+              LOG.warn("Starting MOB cleaning cycle from the beginning due to error:",e);
+              continue;
+            }
+            succeed = true;
+          }
+          // Add MOB refs for current region/family
+          allActiveMobFileName.addAll(regionMobs);
+        } // END column families
+      } // END regions
+      // Check if number of MOB files too big (over 1M)
+      if (allActiveMobFileName.size() > 1000000) {
+        LOG.warn("Found too many active MOB files: {}, this may result in high memory pressure.",
+          allActiveMobFileName.size());
+      }
+      LOG.debug("Found: {} active mob refs", allActiveMobFileName.size());
+      allActiveMobFileName.stream().forEach(LOG::debug);
+
+      // Now scan MOB directories and find MOB files with no references to them
+      for (ColumnFamilyDescriptor hcd : list) {
+        List<Path> toArchive = new ArrayList<Path>();
+        String family = hcd.getNameAsString();
+        Path dir = MobUtils.getMobFamilyPath(conf, table, family);
+        RemoteIterator<LocatedFileStatus> rit = fs.listLocatedStatus(dir);
+        while (rit.hasNext()) {
+          LocatedFileStatus lfs = rit.next();
+          Path p = lfs.getPath();
+          if (!allActiveMobFileName.contains(p.getName())) {
+            // MOB is not in a list of active references, but it can be too
+            // fresh, skip it in this case
+            long creationTime = fs.getFileStatus(p).getModificationTime();
+            if ( creationTime < maxCreationTimeToArchive) {
+              /* DEBUG */ LOG.info(
+                " Archiving MOB file{} creation time=" + (fs.getFileStatus(p).getModificationTime()), p);
+              toArchive.add(p);
+            } else {
+              LOG.info("Skipping fresh file: {}", p);
+            }
+          } else {
+            LOG.info("Keepeing active MOB file: {}", p);
+          }
+        }
+        LOG.info(" MOB Cleaner found {} files for family={}", toArchive.size() , family);
```

Comment:
include table name and specify that it's files to archive

Reply:
Fixed.